modules/graph_ql/main.tf

```diff
@@ -1,4 +1,133 @@
-data "aws_iam_policy_document" "graphql_task_execution_policy_document" {
+# Local Variables
+locals {
+  graphql_port = 5000
+}
+
+resource "random_id" "lb_name" {
+  byte_length = 4
+}
+
+resource "aws_security_group" "gql_security_group" {
+  name        = "${var.prefix}-gql"
+  description = "Allow inbound communication on container port."
+  vpc_id      = var.vpc_id
+
+  ingress {
+    description      = "Container port communication."
+    from_port        = local.graphql_port
+    to_port          = local.graphql_port
+    protocol         = "tcp"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = var.tags
+}
+
+# Application load balancer
+resource "aws_lb" "gql_app_lb" {
+  name               = "${var.prefix}-gql-a" # Max 32 characters. Some prefixes are 25 characters long.
+  internal           = true
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.gql_security_group.id]
+  subnets            = var.lambda_subnet_ids
+  idle_timeout       = 30 # API Gateway locks us to 30 seconds.
+  tags               = var.tags
+}
+
+data "aws_lb" "gql_app_lb_data" {
+  arn = aws_lb.gql_app_lb.arn
+}
+
+resource "aws_lb_target_group" "gql_app_lb_target_group" {
+  name        = "${random_id.lb_name.hex}-gql-a" # name must be randomized. Max 32 characters. Some prefixes are 25 characters long.
+  vpc_id      = var.vpc_id
+  protocol    = "HTTP"
+  port        = local.graphql_port
+  target_type = "ip"
+
+  # NOTE: TF is unable to destroy a target group while a listener is attached,
+  # therefore we have to create a new one before destroying the old. This also means
+  # we have to let it have a random name.
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  health_check {
+    path = "/healthz"
+  }
+
+  tags = var.tags
+}
+
+resource "aws_lb_listener" "gql_app_lb_listener" {
+  load_balancer_arn = aws_lb.gql_app_lb.arn
+  port              = local.graphql_port
+  protocol          = "HTTP"
```

Comment:
I think `HTTPS ` should be more secured.

Reply:
Agreed. Sadly, there are some additional requirements that come with that. https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_listener#certificate_arn